src/dbnode/storage/index/compaction/compactor.go

```diff
@@ -250,6 +254,36 @@ func (c *Compactor) compactFromBuilderWithLock(
 		}
 	}()
 
+	if !c.opts.MmapDocsData {
+		// If retaining references to the original docs, simply take ownership
+		// of the documents and then reference them directly from the FST segment
+		// rather than encoding them and mmap'ing the encoded documents.
+		allDocsCopy := make([]doc.Document, len(allDocs))
+		copy(allDocsCopy, allDocs)
+		fstData.DocsReader = docs.NewSliceReader(0, allDocsCopy)
+	} else {
+		// Otherwise encode and reference the encoded bytes as mmap'd bytes.
```

Comment:
I'm not really familiar with this path, but just verifying it's ok to have DocsReader be nil here?

Reply:
Yup, if we supply the docs data and docs index data then its safe to have DocsReader be nil.